java/vector/src/main/java/org/apache/arrow/vector/complex/RunEndEncodedVector.java

```
@@ -0,0 +1,670 @@
+   *     target vector of the same type.
+   */
+  @Override
+  public TransferPair getTransferPair(String ref, BufferAllocator allocator, CallBack callBack) {
+    throw new UnsupportedOperationException(
+        "RunEndEncodedVector does not support getTransferPair(String, BufferAllocator, CallBack)");
+  }
+
+  /**
+   * To transfer quota responsibility.
+   *
+   * @param field the Field object used by the target vector
+   * @param allocator the target allocator
+   * @param callBack A schema change callback.
+   * @return a {@link org.apache.arrow.vector.util.TransferPair transfer pair}, creating a new
+   *     target vector of the same type.
+   */
+  @Override
+  public TransferPair getTransferPair(Field field, BufferAllocator allocator, CallBack callBack) {
+    throw new UnsupportedOperationException(
+        "RunEndEncodedVector does not support getTransferPair(Field, BufferAllocator, CallBack)");
+  }
+
+  /**
+   * Makes a new transfer pair used to transfer underlying buffers.
+   *
+   * @param target the target for the transfer
+   * @return a new {@link org.apache.arrow.vector.util.TransferPair transfer pair} that is used to
+   *     transfer underlying buffers into the target vector.
+   */
+  @Override
+  public TransferPair makeTransferPair(ValueVector target) {
+    throw new UnsupportedOperationException(
+        "RunEndEncodedVector does not support makeTransferPair(ValueVector)");
+  }
+
+  /**
+   * Get a reader for this vector.
+   *
+   * @return a {@link org.apache.arrow.vector.complex.reader.FieldReader field reader} that supports
+   *     reading values from this vector.
+   */
+  @Override
+  public FieldReader getReader() {
+    return null; // TODO
+  }
+
+  /**
+   * Get a writer for this vector.
+   *
+   * @return a {@link org.apache.arrow.vector.complex.writer.FieldWriter field writer} that supports
+   *     writing values to this vector.
+   */
+  public FieldWriter getWriter() {
+    return null; // TODO
```

Same here and elsewhere, throw instead of returning null